hudi-client/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -117,7 +118,17 @@ public boolean commitStats(String instantTime, List<HoodieWriteStat> stats, Opti
     if (extraMetadata.isPresent()) {
       extraMetadata.get().forEach(metadata::addMetadata);
     }
-    metadata.addMetadata(HoodieCommitMetadata.SCHEMA_KEY, config.getSchema());
+    String schema = config.getSchema();
+    if (config.updatePartialFields()) {
```

Comment:
Can we have this so that whether or not we can update partial fields is simply controlled by the payload impl. 
I am concerned that a config like `config.updatePartialFields` is not inline with how we treat payloads. for e.g for MOR, there is no writeConfig when we merge the fields via record reader